.github/workflows/test.yml

```diff
@@ -25,52 +25,123 @@ jobs:
+            wagtail: "wagtail>=4.2,<5.0"
+            database: "sqlite3"
+            experimental: false
+
           - python: 3.9
-            django: "Django>=3.2,<3.3"
-            wagtail: "wagtail>=2.14,<2.15"
+            django: "Django>=3.2,<4.0"
+            wagtail: "wagtail>=4.1,<4.2"
             database: "postgresql"
             experimental: false
           - python: 3.9
             django: "Django>=3.2,<4.0"
-            wagtail: "wagtail>=2.15,<2.16"
+            wagtail: "wagtail>=4.2,<5.0"
+            database: "sqlite3"
+            experimental: false
+          - python: 3.9
+            django: "Django>=4.0,<4.1"
+            wagtail: "wagtail>=4.1,<4.2"
+            database: "postgresql"
+            experimental: false
+          - python: 3.9
+            django: "Django>=4.0,<4.1"
+            wagtail: "wagtail>=4.2,<5.0"
             database: "sqlite3"
             experimental: false
+          - python: 3.9
+            django: "Django>=4.1,<4.2"
+            wagtail: "wagtail>=4.1,<4.2"
+            database: "postgresql"
+            experimental: false
+          - python: 3.9
+            django: "Django>=4.1,<4.2"
+            wagtail: "wagtail>=4.2,<5.0"
+            database: "sqlite3"
+            experimental: false
+
+          - python: "3.10"
+            django: "Django>=3.2,<4.0"
+            wagtail: "wagtail>=4.1,<4.2"
+            database: "postgresql"
+            experimental: false
           - python: "3.10"
-            django: "Django>=3.2,<4.1"
-            wagtail: "wagtail>=2.16,<3.0"
+            django: "Django>=3.2,<4.0"
+            wagtail: "wagtail>=4.2,<5.0"
             database: "sqlite3"
             experimental: false
           - python: "3.10"
-            django: "Django>=3.2,<4.1"
-            wagtail: "wagtail>=3.0,<4.0"
+            django: "Django>=4.0,<4.1"
+            wagtail: "wagtail>=4.1,<4.2"
             database: "postgresql"
             experimental: false
           - python: "3.10"
-            django: "Django>=3.2,<4.2"
-            wagtail: "wagtail>=4.0,<5.0"
+            django: "Django>=4.0,<4.1"
+            wagtail: "wagtail>=4.2,<5.0"
+            database: "sqlite3"
+            experimental: false
+          - python: "3.10"
+            django: "Django>=4.1,<4.2"
+            wagtail: "wagtail>=4.1,<4.2"
             database: "postgresql"
             experimental: false
           - python: "3.10"
-            django: "Django>=3.2,<4.2"
+            django: "Django>=4.1,<4.2"
+            wagtail: "wagtail>=4.2,<5.0"
+            database: "sqlite3"
+            experimental: false
+
+          - python: "3.11"
+            django: "Django>=4.1,<4.2"
+            wagtail: "wagtail>=4.1,<4.2"
+            database: "postgresql"
+            experimental: false
+          - python: "3.11"
+            django: "Django>=4.1,<4.2"
+            wagtail: "wagtail>=4.2,<5.0"
+            database: "sqlite3"
+            experimental: false
```

Comment:
I think this can be trimmed down a bit to save on CI resources / runtime - we don't really need to test every permutation of Python, Django and Wagtail. Also, the comment at the top of this section needs to be updated to match any changes.